### ISSUE TITLE
`check_outliers()` example doesn't work

### ISSUE THREAD

**Issue by IndrajeetPatil** [easystats/performance#555]
Seen here:

https://github.com/easystats/performance/blob/c664950ec73186f7774ece024ca5a6bf642d27ec/R/check_outliers.R#L291-L295

Reprex:

``` r
library(performance)
library(dplyr)

iris %>%
  group_by(Species) %>%
  check_outliers()
#> Error in `[[<-`:
#> ! Can't convert `value` <factor<fb977>> to <integer>.
#> Backtrace:
#>      ▆
#>   1. ├─iris %>% group_by(Species) %>% check_outliers()
#>   2. ├─performance::check_outliers(.)
#>   3. ├─performance:::check_outliers.grouped_df(.)
#>   4. │ └─base::lapply(...)
#>   5. │   └─performance (local) FUN(X[[i]], ...)
#>   6. │     ├─base::`[[<-`(`*tmp*`, x, value = `<fct>`)
#>   7. │     └─vctrs:::`[[<-.vctrs_list_of`(`*tmp*`, x, value = `<fct>`)
#>   8. │       └─vctrs::vec_cast(value, attr(x, "ptype"))
#>   9. └─vctrs (local) `<fn>`()
#>  10.   └─vctrs::vec_default_cast(...)
#>  11.     ├─base::withRestarts(...)
#>  12.     │ └─base (local) withOneRestart(expr, restarts[[1L]])
#>  13.     │   └─base (local) doWithOneRestart(return(expr), restart)
#>  14.     └─vctrs::stop_incompatible_cast(...)
#>  15.       └─vctrs::stop_incompatible_type(...)
#>  16.         └─vctrs:::stop_incompatible(...)
#>  17.           └─vctrs:::stop_vctrs(...)
#>  18.             └─rlang::abort(message, class = c(class, "vctrs_error"), ..., call = call)
```

<sup>Created on 2023-03-17 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

<details style="margin-bottom:10px;">
<summary>
Session info
</summary>

``` r
sessioninfo::session_info()
#> ─ Session info ───────────────────────────────────────────────────────────────
#>  setting  value
#>  version  R version 4.2.2 (2022-10-31)
#>  os       macOS Ventura 13.1
#>  system   aarch64, darwin20
#>  ui       X11
#>  language (EN)
#>  collate  en_US.UTF-8
#>  ctype    en_US.UTF-8
#>  tz       Europe/Berlin
#>  date     2023-03-17
#>  pandoc   3.0 @ /usr/local/bin/ (via rmarkdown)
#> 
#> ─ Packages ───────────────────────────────────────────────────────────────────
#>  package     * version    date (UTC) lib source
#>  cli           3.6.0      2023-01-09 [1] CRAN (R 4.2.2)
#>  datawizard    0.6.5.7    2023-02-20 [1] https://easystats.r-universe.dev (R 4.2.2)
#>  digest        0.6.31     2022-12-11 [1] CRAN (R 4.2.2)
#>  dplyr       * 1.1.0      2023-01-29 [1] CRAN (R 4.2.0)
#>  evaluate      0.20       2023-01-17 [1] CRAN (R 4.2.0)
#>  fansi         1.0.4      2023-01-22 [1] CRAN (R 4.2.0)
#>  fastmap       1.1.1      2023-02-24 [1] CRAN (R 4.2.0)
#>  fs            1.6.1      2023-02-06 [1] CRAN (R 4.2.0)
#>  generics      0.1.3      2022-07-05 [1] CRAN (R 4.2.1)
#>  glue          1.6.2      2022-02-24 [1] CRAN (R 4.2.0)
#>  htmltools     0.5.4      2022-12-07 [1] CRAN (R 4.2.2)
#>  insight       0.19.0.2   2023-02-22 [1] https://easystats.r-universe.dev (R 4.2.2)
#>  knitr         1.42       2023-01-25 [1] CRAN (R 4.2.2)
#>  lifecycle     1.0.3      2022-10-07 [1] CRAN (R 4.2.1)
#>  magrittr      2.0.3      2022-03-30 [1] CRAN (R 4.2.0)
#>  performance * 0.10.2.6   2023-03-17 [1] https://easystats.r-universe.dev (R 4.2.2)
#>  pillar        1.8.1      2022-08-19 [1] CRAN (R 4.2.1)
#>  pkgconfig     2.0.3      2019-09-22 [1] CRAN (R 4.2.0)
#>  purrr         1.0.1      2023-01-10 [1] CRAN (R 4.2.2)
#>  R.cache       0.16.0     2022-07-21 [1] CRAN (R 4.2.0)
#>  R.methodsS3   1.8.2      2022-06-13 [1] CRAN (R 4.2.0)
#>  R.oo          1.25.0     2022-06-12 [1] CRAN (R 4.2.0)
#>  R.utils       2.12.2     2022-11-11 [1] CRAN (R 4.2.2)
#>  R6            2.5.1.9000 2022-10-27 [1] local
#>  reprex        2.0.2      2022-08-17 [1] CRAN (R 4.2.1)
#>  rlang         1.1.0      2023-03-14 [1] CRAN (R 4.2.0)
#>  rmarkdown     2.20       2023-01-19 [1] CRAN (R 4.2.2)
#>  rstudioapi    0.14       2022-08-22 [1] CRAN (R 4.2.1)
#>  sessioninfo   1.2.2      2021-12-06 [1] CRAN (R 4.2.0)
#>  styler        1.9.1      2023-03-04 [1] CRAN (R 4.2.2)
#>  tibble        3.2.0      2023-03-08 [1] CRAN (R 4.2.0)
#>  tidyselect    1.2.0      2022-10-10 [1] CRAN (R 4.2.1)
#>  utf8          1.2.3      2023-01-31 [1] CRAN (R 4.2.2)
#>  vctrs         0.6.0      2023-03-16 [1] CRAN (R 4.2.2)
#>  withr         2.5.0      2022-03-03 [1] CRAN (R 4.2.0)
#>  xfun          0.37       2023-01-31 [1] CRAN (R 4.2.0)
#>  yaml          2.3.7      2023-01-23 [1] CRAN (R 4.2.2)
#> 
#>  [1] /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/library
#> 
#> ──────────────────────────────────────────────────────────────────────────────
```

</details>

cc @rempsyc

**Comment by strengejacke** [comment]
Is this issue resolved?

**Comment by IndrajeetPatil** [comment]
Not yet. I still the same issue.

**Comment by strengejacke** [comment]
Seems to work now, at least locally. Will wait for GH Actions example checks...

**Comment by rempsyc** [comment]
This is what I'm getting: works with `poorman`, not with `dplyr`:

``` r
library(performance)

iris |>
  poorman::group_by(Species) |>
  check_outliers()
#> OK: No outliers detected.
#> - Based on the following method and threshold: mahalanobis (18.47).
#> - For variables: Sepal.Length, Sepal.Width, Petal.Length, Petal.Width

iris |>
  dplyr::group_by(Species) |>
  check_outliers()
#> Error in `[[<-` at performance/R/check_outliers.R:1141:4:
#> ! Can't convert `value` <factor<fb977>> to <integer>.
#> Backtrace:
#>      ▆
#>   1. ├─performance::check_outliers(dplyr::group_by(iris, Species))
#>   2. ├─performance:::check_outliers.grouped_df(...) at performance/R/check_outliers.R:309:2
#>   3. │ └─base::lapply(...) at performance/R/check_outliers.R:1140:2
#>   4. │   └─performance (local) FUN(X[[i]], ...)
#>   5. │     ├─base::`[[<-`(`*tmp*`, x, value = `<fct>`) at performance/R/check_outliers.R:1141:4
#>   6. │     └─vctrs:::`[[<-.vctrs_list_of`(`*tmp*`, x, value = `<fct>`) at performance/R/check_outliers.R:1141:4
#>   7. │       └─vctrs::vec_cast(value, attr(x, "ptype"))
#>   8. └─vctrs (local) `<fn>`()
#>   9.   └─vctrs::vec_default_cast(...)
#>  10.     ├─base::withRestarts(...)
#>  11.     │ └─base (local) withOneRestart(expr, restarts[[1L]])
#>  12.     │   └─base (local) doWithOneRestart(return(expr), restart)
#>  13.     └─vctrs::stop_incompatible_cast(...)
#>  14.       └─vctrs::stop_incompatible_type(...)
#>  15.         └─vctrs:::stop_incompatible(...)
#>  16.           └─vctrs:::stop_vctrs(...)
#>  17.             └─rlang::abort(message, class = c(class, "vctrs_error"), ..., call = call)

packageVersion("poorman")
#> [1] '0.2.6'
packageVersion("dplyr")
#> [1] '1.1.1'
```

<sup>Created on 2023-03-26 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by IndrajeetPatil** [comment]
Let's ask our resident expert on `{poorman}`-`{dplyr}` interaction. Pinging @etiennebacher...

**Comment by etiennebacher** [comment]
Well...

``` r
library(performance)

packageVersion("performance")
#> [1] '0.10.2.7'
packageVersion("poorman")
#> [1] '0.2.6'
packageVersion("dplyr")
#> [1] '1.1.1'
packageVersion("vctrs")
#> [1] '0.6.1'

iris |>
  poorman::group_by(Species) |>
  check_outliers()
#> OK: No outliers detected.
#> - Based on the following method and threshold: mahalanobis (20).
#> - For variables: Sepal.Length, Sepal.Width, Petal.Length, Petal.Width

iris |>
  dplyr::group_by(Species) |>
  check_outliers()
#> OK: No outliers detected.
#> - Based on the following method and threshold: mahalanobis (20).
#> - For variables: Sepal.Length, Sepal.Width, Petal.Length, Petal.Width
```

Did you update all your packages?

**Comment by IndrajeetPatil** [comment]
Instead of namesping, if you run _exactly_ the following code, what happens?

```r
library(performance)
library(dplyr)

iris %>%
  group_by(Species) %>%
  check_outliers()
```

**Comment by IndrajeetPatil** [comment]
Whaaaa!!! I have the same package versions!

``` r
library(performance)

packageVersion("performance")
#> [1] '0.10.2.7'
packageVersion("poorman")
#> [1] '0.2.6'
packageVersion("dplyr")
#> [1] '1.1.1'
packageVersion("vctrs")
#> [1] '0.6.1'

iris |>
  poorman::group_by(Species) |>
  check_outliers()
#> OK: No outliers detected.
#> - Based on the following method and threshold: mahalanobis (20).
#> - For variables: Sepal.Length, Sepal.Width, Petal.Length, Petal.Width

iris |>
  dplyr::group_by(Species) |>
  check_outliers()
#> Error in `[[<-`:
#> ! Can't convert `value` <factor<fb977>> to <integer>.
#> Backtrace:
#>      ▆
#>   1. ├─performance::check_outliers(dplyr::group_by(iris, Species))
#>   2. ├─performance:::check_outliers.grouped_df(...)
#>   3. │ └─base::lapply(...)
#>   4. │   └─performance (local) FUN(X[[i]], ...)
#>   5. │     ├─base::`[[<-`(`*tmp*`, x, value = `<fct>`)
#>   6. │     └─vctrs:::`[[<-.vctrs_list_of`(`*tmp*`, x, value = `<fct>`)
#>   7. │       └─vctrs::vec_cast(value, attr(x, "ptype"))
#>   8. └─vctrs (local) `<fn>`()
#>   9.   └─vctrs::vec_default_cast(...)
#>  10.     ├─base::withRestarts(...)
#>  11.     │ └─base (local) withOneRestart(expr, restarts[[1L]])
#>  12.     │   └─base (local) doWithOneRestart(return(expr), restart)
#>  13.     └─vctrs::stop_incompatible_cast(...)
#>  14.       └─vctrs::stop_incompatible_type(...)
#>  15.         └─vctrs:::stop_incompatible(...)
#>  16.           └─vctrs:::stop_vctrs(...)
#>  17.             └─rlang::abort(message, class = c(class, "vctrs_error"), ..., call = call)
```

<sup>Created on 2023-03-27 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

<details style="margin-bottom:10px;">
<summary>
Session info
</summary>

``` r
sessioninfo::session_info()
#> ─ Session info ───────────────────────────────────────────────────────────────
#>  setting  value
#>  version  R version 4.2.2 (2022-10-31)
#>  os       macOS Ventura 13.2.1
#>  system   aarch64, darwin20
#>  ui       X11
#>  language (EN)
#>  collate  en_US.UTF-8
#>  ctype    en_US.UTF-8
#>  tz       Europe/Berlin
#>  date     2023-03-27
#>  pandoc   3.1.1 @ /Applications/RStudio.app/Contents/Resources/app/quarto/bin/tools/ (via rmarkdown)
#> 
#> ─ Packages ───────────────────────────────────────────────────────────────────
#>  package     * version    date (UTC) lib source
#>  cli           3.6.1      2023-03-23 [1] CRAN (R 4.2.2)
#>  datawizard    0.7.0.2    2023-03-22 [1] local
#>  digest        0.6.31     2022-12-11 [1] CRAN (R 4.2.2)
#>  dplyr         1.1.1      2023-03-22 [1] CRAN (R 4.2.2)
#>  evaluate      0.20       2023-01-17 [1] CRAN (R 4.2.0)
#>  fansi         1.0.4      2023-01-22 [1] CRAN (R 4.2.0)
#>  fastmap       1.1.1      2023-02-24 [1] CRAN (R 4.2.0)
#>  fs            1.6.1      2023-02-06 [1] CRAN (R 4.2.0)
#>  generics      0.1.3      2022-07-05 [1] CRAN (R 4.2.1)
#>  glue          1.6.2      2022-02-24 [1] CRAN (R 4.2.0)
#>  htmltools     0.5.5      2023-03-23 [1] CRAN (R 4.2.2)
#>  insight       0.19.1.2   2023-03-25 [1] Github (easystats/insight@d63f2aa)
#>  knitr         1.42       2023-01-25 [1] CRAN (R 4.2.2)
#>  lifecycle     1.0.3      2022-10-07 [1] CRAN (R 4.2.1)
#>  magrittr      2.0.3      2022-03-30 [1] CRAN (R 4.2.0)
#>  performance * 0.10.2.7   2023-03-25 [1] Github (easystats/performance@6e4a1b7)
#>  pillar        1.9.0      2023-03-22 [1] CRAN (R 4.2.2)
#>  pkgconfig     2.0.3      2019-09-22 [1] CRAN (R 4.2.0)
#>  poorman       0.2.6      2022-08-16 [1] Github (nathaneastwood/poorman@d7d73db)
#>  purrr         1.0.1      2023-01-10 [1] CRAN (R 4.2.2)
#>  R.cache       0.16.0     2022-07-21 [1] CRAN (R 4.2.0)
#>  R.methodsS3   1.8.2      2022-06-13 [1] CRAN (R 4.2.0)
#>  R.oo          1.25.0     2022-06-12 [1] CRAN (R 4.2.0)
#>  R.utils       2.12.2     2022-11-11 [1] CRAN (R 4.2.2)
#>  R6            2.5.1.9000 2022-10-27 [1] local
#>  reprex        2.0.2      2022-08-17 [1] CRAN (R 4.2.1)
#>  rlang         1.1.0      2023-03-14 [1] CRAN (R 4.2.0)
#>  rmarkdown     2.21       2023-03-26 [1] CRAN (R 4.2.2)
#>  rstudioapi    0.14       2022-08-22 [1] CRAN (R 4.2.1)
#>  sessioninfo   1.2.2      2021-12-06 [1] CRAN (R 4.2.0)
#>  styler        1.9.1      2023-03-04 [1] CRAN (R 4.2.2)
#>  tibble        3.2.1      2023-03-20 [1] CRAN (R 4.2.2)
#>  tidyselect    1.2.0      2022-10-10 [1] CRAN (R 4.2.1)
#>  utf8          1.2.3      2023-01-31 [1] CRAN (R 4.2.2)
#>  vctrs         0.6.1      2023-03-22 [1] CRAN (R 4.2.2)
#>  withr         2.5.0      2022-03-03 [1] CRAN (R 4.2.0)
#>  xfun          0.38       2023-03-24 [1] CRAN (R 4.2.2)
#>  yaml          2.3.7      2023-01-23 [1] CRAN (R 4.2.2)
#> 
#>  [1] /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/library
#> 
#> ──────────────────────────────────────────────────────────────────────────────
```

</details>

**Comment by etiennebacher** [comment]
Can you do `pak::pkg_deps_tree("dplyr")` to show the versions of all deps of `dplyr`?

``` r
pak::pkg_deps_tree("dplyr")
#> ℹ Loading metadata database✔ Loading metadata database ... done
#> dplyr 1.1.1 [new][dl] (1.55 MB)
#> ├─cli 3.6.1 [new][dl] (1.30 MB)
#> ├─generics 0.1.3 [new][dl] (80.61 kB)
#> ├─glue 1.6.2 [new][dl] (162.17 kB)
#> ├─lifecycle 1.0.3 [new][dl] (138.99 kB)
#> │ ├─cli
#> │ ├─glue
#> │ └─rlang 1.1.0 [new][dl] (1.58 MB)
#> ├─magrittr 2.0.3 [new][dl] (226.95 kB)
#> ├─pillar 1.9.0 [new][dl] (658.76 kB)
#> │ ├─cli
#> │ ├─fansi 1.0.4 [new][dl] (313.03 kB)
#> │ ├─glue
#> │ ├─lifecycle
#> │ ├─rlang
#> │ ├─utf8 1.2.3 [new][dl] (149.91 kB)
#> │ └─vctrs 0.6.1 [new][dl] (1.34 MB)
#> │   ├─cli
#> │   ├─glue
#> │   ├─lifecycle
#> │   └─rlang
#> ├─R6 2.5.1 [new][dl] (84.28 kB)
#> ├─rlang
#> ├─tibble 3.2.1 [new][dl] (690.25 kB)
#> │ ├─fansi
#> │ ├─lifecycle
#> │ ├─magrittr
#> │ ├─pillar
#> │ ├─pkgconfig 2.0.3 [new][dl] (22.42 kB)
#> │ ├─rlang
#> │ └─vctrs
#> ├─tidyselect 1.2.0 [new][dl] (224.07 kB)
#> │ ├─cli
#> │ ├─glue
#> │ ├─lifecycle
#> │ ├─rlang
#> │ ├─vctrs
#> │ └─withr 2.5.0 [new][dl] (231.81 kB)
#> └─vctrs
#> 
#> Key:  [new] new | [dl] download
```

<sup>Created on 2023-03-27 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by IndrajeetPatil** [comment]
Actually, the other relevant difference I see between our sessions is the followjng:

```r
# EB
#>  vctrs         0.6.1      2023-03-22 [1] CRAN (R 4.2.3)

# IP
#>  vctrs         0.6.1      2023-03-22 [1] CRAN (R 4.2.2)
```

**Comment by IndrajeetPatil** [comment]
```r
> pak::pkg_deps_tree("dplyr")
✔ Updated metadata database: 4.34 MB in 5 files.                          
✔ Updating metadata database ... done                                     
dplyr 1.1.1 ✨ ⬇ (1.57 MB)                                                  
├─cli 3.6.1 ✨ ⬇ (1.37 MB)
├─generics 0.1.3 ✨ ⬇ (78.24 kB)
├─glue 1.6.2 ✨ ⬇ (154.64 kB)
├─lifecycle 1.0.3 ✨ ⬇ (122.95 kB)
│ ├─cli
│ ├─glue
│ └─rlang 1.1.0 ✨ ⬇ (1.87 MB)
├─magrittr 2.0.3 ✨ ⬇ (231.25 kB)
├─pillar 1.9.0 ✨ ⬇ (645.16 kB)
│ ├─cli
│ ├─fansi 1.0.4 ✨ ⬇ (379.60 kB)
│ ├─glue
│ ├─lifecycle
│ ├─rlang
│ ├─utf8 1.2.3 ✨ ⬇ (208.79 kB)
│ └─vctrs 0.6.1 ✨ ⬇ (1.88 MB)
│   ├─cli
│   ├─glue
│   ├─lifecycle
│   └─rlang
├─R6 2.5.1 ✨ ⬇ (82.52 kB)
├─rlang
├─tibble 3.2.1 ✨ ⬇ (681.62 kB)
│ ├─fansi
│ ├─lifecycle
│ ├─magrittr
│ ├─pillar
│ ├─pkgconfig 2.0.3 ✨ ⬇ (17.70 kB)
│ ├─rlang
│ └─vctrs
├─tidyselect 1.2.0 ✨ ⬇ (220.69 kB)
│ ├─cli
│ ├─glue
│ ├─lifecycle
│ ├─rlang
│ ├─vctrs
│ └─withr 2.5.0 ✨ ⬇ (229.48 kB)
└─vctrs

Key:  ✨ new |  ⬇ download
```

**Comment by etiennebacher** [comment]
I don't see any diff, two potential causes:

* different R versions 
* different OS

Both quite unlikely...

**Comment by rempsyc** [comment]
@IndrajeetPatil I thought I fixed that in #564, not sure why it didn't close this issue after meeting. Are you saying even after downloading the new version with the fix (I didn't bump the version) you are still getting this?

**Comment by IndrajeetPatil** [comment]
Oh, indeed. The issue seems to be fixed now. Thanks!

Sorry, didn't realize that I didn't have the latest _performance_ commit.
